src/System.Configuration.ConfigurationManager/src/System/Configuration/ImplicitMachineConfigHost.cs

```diff
@@ -23,14 +23,14 @@ internal ImplicitMachineConfigHost(IInternalConfigHost host)
             out string locationConfigPath, IInternalConfigRoot configRoot, params object[] hostInitConfigurationParams)
         {
             // Stash the filemap so we can see if the machine config was explicitly specified
-            _fileMap = (ConfigurationFileMap)hostInitConfigurationParams[0];
+            _fileMap = hostInitConfigurationParams[0] as ConfigurationFileMap;
```

Comment:
Is there a test we could add that would have resulted in an exception and now won't?

Reply:
Added a few more files to the PR.  The test passes w/ my change and fails when run against the old way.

Reply:
@stephentoub - any reason why InternalConfigConfigurationFactory would fail in the netfx x86 build?

Reply:
@steveisok the error is `System.TypeLoadException : Could not load type 'System.Configuration.Internal.InternalConfigConfigurationFactory' from assembly 'System.Configuration.ConfigurationManager'.`

On NETFX you need to look for the typ in System.Configuration.dll